src/store/friends/features/get-user-by-phone/get-user-by-phone.ts

```diff
@@ -27,12 +27,14 @@ export class GetUserByPhone {
           .filter((x) => x !== ' ' && x !== '+')
           .join('');
 
-        const { data } = GetUserByPhone.httpRequest.call(
+        const response = GetUserByPhone.httpRequest.call(
           yield call(() => GetUserByPhone.httpRequest.generator({ phone: parsedPhone })),
         );
 
-        action.meta?.deferred?.resolve(data);
-      } catch {
+        if (response.status === 204) throw new Error('204 has been returned');
```

Comment:
regarding performance it will be better to call reject on promise directly instead of throwing error